lib/share_progress/button.rb

```diff
@@ -65,19 +68,31 @@ def initialize(params)
     end
 
     def update_attributes(params)
-      @id = params['id'] if params.include? 'id'
-      self.page_url = params['page_url'] if params.include? 'page_url'
-      self.is_active = params['is_active'] if params.include? 'is_active'
-      self.page_title = params['page_title'] if params.include? 'page_title'
-      self.button_template = params['button_template'] if params.include? 'button_template'
-      self.share_button_html = params['share_button_html'] if params.include? 'share_button_html'
+      params = Utils.symbolize_keys(params)
+      @id = params[:id] if params.include? :id
+      @errors = params[:errors] if params.include? :errors
+      self.page_url = params[:page_url] if params.include? :page_url
```

Comment:
I think the method I've been looking for is `instance_variable_set`. commit forthcoming!